### PR TITLE
Refactored TestTestSuiteMinimizer, TestGenerationContext and EvoSuite

### PR DESCRIPTION
**Pull request by prathmesh-halgekar**

### CODE REVIEW

File: master/src/main/java/org/evosuite/EvoSuite.java [hunk 1]
```diff
@@ -215,9 +215,9 @@ public Object parseCommandLine(String[] args) {
 
             CommandLineParameters.addJavaDOptions(javaOpts, line);
 
-            if (TestSuiteWriterUtils.needToUseAgent() && Properties.JUNIT_CHECK) {
-                ClassPathHacker.initializeToolJar();
-            }
+//            if (TestSuiteWriterUtils.needToUseAgent() && Properties.JUNIT_CHECK) {
+//                ClassPathHacker.initializeToolJar();
```
Comment (gofraser) [inline master/src/main/java/org/evosuite/EvoSuite.java]: I understand that adding tools jar does not work with Java 9. But wouldn't removing it break EvoSuite on Java < 9?

File: client/src/main/resources/java9_invisible_packages.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sun.java2d=sun.java2d
```
Comment (gofraser) [inline client/src/main/resources/java9_invisible_packages.properties]: Why is this a properties file? Wouldn't a plain text file be sufficient? (Like e.g. client/src/main/resources//jdkPureMethods.txt)

File: client/src/main/java/org/evosuite/TestGenerationContext.java [hunk 3]
```diff
@@ -213,20 +213,24 @@ public void resetContext() {
 		CarvingManager.getInstance().clear();
 
 		// TODO: Why are we doing this?
-		if (Properties.INSTRUMENT_CONTEXT || ArrayUtil.contains(Properties.CRITERION, Properties.Criterion.DEFUSE)
+		/*if (Properties.INSTRUMENT_CONTEXT || ArrayUtil.contains(Properties.CRITERION, Properties.Criterion.DEFUSE)
```
Comment (gofraser) [inline client/src/main/java/org/evosuite/TestGenerationContext.java]: Why don't we need this?

File: master/src/main/java/org/evosuite/EvoSuite.java (hunk 1 above)
Comment (prathmesh-halgekar) [reply master/src/main/java/org/evosuite/EvoSuite.java]: That was a redundant check. Line no. 160- 162 in the same file already takes care of it.

File: client/src/main/resources/java9_invisible_packages.properties (hunk 2 above)
Comment (prathmesh-halgekar) [reply client/src/main/resources/java9_invisible_packages.properties]: I guess it makes it easier to read using ResourceBundle.

Comment (gofraser) [reply client/src/main/resources/java9_invisible_packages.properties]: But it's a bit weird, you need to specify every package twice as key=value with key being the same as value.

Comment (prathmesh-halgekar) [reply client/src/main/resources/java9_invisible_packages.properties]: Refactored to use java9_invisible_packages.txt

**Pull request merged**